src/tests/system/tests/test_ldap_extra_attrs.py

```diff
@@ -42,3 +48,198 @@ def test_ldap_extra_attrs__filled(client: Client, provider: GenericProvider, att
+    client.sssd.sssd["services"] = "nss, pam, ifp"
+    client.sssd.start()
+
+    result = client.tools.getent.passwd("user1")
+    assert result is not None, "User is not found"
+    assert result.name == "user1", "User has wrong name"
+
+    search = client.ldb.search(
+        f"/var/lib/sss/db/cache_{client.sssd.default_domain}.ldb", f"cn=users,cn={client.sssd.default_domain},cn=sysdb"
+    )
+
+    user_dict = search["name=user1@test,cn=users,cn=test,cn=sysdb"]
+    assert user_dict["description"] == ["gecos1"], "attribute 'descripion' was not correct"
+    assert user_dict["mail"] == ["user1@example.test"], "attribute 'mail' was not correct"
+    assert user_dict["email"] == ["user1@example.test"], "attribute 'email' was not correct"
+
+
+@pytest.mark.ticket(bz=1667252)
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_ldap_extra_attrs__ifp(client: Client, provider: GenericProvider):
+    """
+    :title: ifp do not crash when requesting extra attributes
+    :setup:
+        1. Create new user "user1"
+        2. Configure properly sssd.conf
+        3. Start SSSD
+    :steps:
+        1. Run "sssctl user-checks user1"
+        2. Check SSSD status
+    :expectedresults:
+        1. Command succeeded
+        2. Checked successfully
+    :customerscenario: True
+    """
+    provider.user("user1").add()
+    client.sssd.sssd["services"] = "nss, pam, ifp"
+    client.sssd.domain["ldap_user_extra_attrs"] = "test:homeDirectory"
+    client.sssd.ifp["user_attributes"] = "+test"
+    client.sssd.start()
+
+    result = client.sssctl.user_checks("user1")
+    assert result.rc == 0, "sssctl user-checks command failed"
+
+    result = client.sssd.svc.status("sssd")
+    assert result.rc == 0, "service status sssd failed"
+
+
+@pytest.mark.ticket(bz=847043)
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_ldap_extra_attrs__thread_issue(client: Client, provider: GenericProvider):
+    """
+    :title: Thread issue can cause the application to not get any identity information
+    :setup:
+        1. Create new user "user1"
+        2. Configure properly sssd.conf
+        3. Start SSSD
+    :steps:
+        1. Start new thread with provided function
+        2. Main thread waits for worker and then sleeps
+        3. Worker is in loop calling getpwuid()
+        4. Main thread cancel worker thread (with exit_flag.set())
+        5. Thread join
+        6. Call getpwuid()
+    :expectedresults:
+        1. Thread is started
+        2. Success
+        3. Success
+        4. Worker is cancelled
+        5. Thread is joined
+        6. Call is not stucked
+    :customerscenario: True
+    """
+
+    def worker():
+        nonlocal started, exit_flag
+        uid = os.geteuid()
+        started.set()
+        while not exit_flag.is_set():
+            pwd.getpwuid(uid)
+
+    provider.user("user1").add()
+    client.sssd.nss["filter_groups"] = "root"
+    client.sssd.nss["filter_users"] = "root"
```

Comment:
I wonder if these `filter*` options are really needed.

Reply:
I am not sure either. I guess they are part of the bugzilla? Could you take a look? I am unable to access this bugzilla. https://bugzilla.redhat.com/show_bug.cgi?id=847043

Reply:
Please try removing them, I don't think these lines are doing anything relevant to the test.

Reply:
I removed them